pkg/gpu/consumer.go

```diff
@@ -0,0 +1,166 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package gpu
+
+import (
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/gpu/model"
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	gpuebpf "github.com/DataDog/datadog-agent/pkg/gpu/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/process/monitor"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// CudaEventConsumer is responsible for consuming CUDA events from the eBPF probe, and delivering them
+// to the appropriate stream handler.
+type CudaEventConsumer struct {
+	eventHandler   ddebpf.EventHandler
+	once           sync.Once
+	closed         chan struct{}
+	streamHandlers map[model.StreamKey]*StreamHandler
+	wg             sync.WaitGroup
+}
+
+// NewCudaEventConsumer creates a new CUDA event consumer.
+func NewCudaEventConsumer(eventHandler ddebpf.EventHandler) *CudaEventConsumer {
+	return &CudaEventConsumer{
+		eventHandler:   eventHandler,
+		closed:         make(chan struct{}),
+		streamHandlers: make(map[model.StreamKey]*StreamHandler),
+	}
+}
+
+// Stop stops the CUDA event consumer.
+func (c *CudaEventConsumer) Stop() {
+	if c == nil {
+		return
+	}
+	c.eventHandler.Stop()
+	c.wg.Wait()
+	c.once.Do(func() {
+		close(c.closed)
+	})
+}
+
+// Start starts the CUDA event consumer.
+func (c *CudaEventConsumer) Start() {
+	if c == nil {
+		return
+	}
+	health := health.RegisterLiveness("gpu-tracer-cuda-events")
+	processMonitor := monitor.GetProcessMonitor()
+	cleanupExit := processMonitor.SubscribeExit(c.handleProcessExit)
+
+	c.wg.Add(1)
+	go func() {
+		processSync := time.NewTicker(5 * time.Second)
```

Comment:
why 5?

Reply:
There's not really a lot of testing, the idea is just to clean up stream handlers for terminated processes every once in a while. I changed the code to have it as a config option in any case.